### DISCUSSION TITLE
Couchdb Replication Sync

### DISCUSSION THREAD

**Discussion by handerson-caylent** [apache/couchdb#4281] [Q&A] [answered]
Hi everyone, 
I'm going to migrate a CouchDB v. 1.6 to a new host, and the plan is to use replication for that... 
However, I couldn't find a way to know when both databases are in sync. The only option I see at the moment is to add a new document in the source DB and wait until it arrives at the target. 
Is there any better way to do that?

Thanks!

**Comment by nickva** [comment, marked as answer]
You could run a pair of replication jobs. A one-shot replication from the source to the new target. Then, make the target the default (production) db. Then perform another one-shot replication from source to target to bring over any document which may have been written there since the start of the previous one, so you don't leave any documents behind. The one-shot (`continuous = false`) replication will just finish when there are no more document left to replicate.

One subtle moment to keep in mind is that one-shot replication replicate from a snapshot of the database from when they start. So it *might not* see any documents written since then.

**Comment by handerson-caylent** [comment]
Great idea, @nickva. As the database is a bit large (around 300GB) and the 1st one-shot replication takes a lot of time, I'll manually move it via SCP, which is faster, and then run a second one-shot replication to push any new document, as you suggested. I think that should work.
Thanks a lot
